action/tx_dynamic_fee.go

```diff
@@ -0,0 +1,141 @@
+package action
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/pkg/errors"
+)
+
+// DynamicFeeTx is a transaction type with dynamic fee cap and tip cap
+type DynamicFeeTx struct {
+	chainID    uint32
+	nonce      uint64
+	gasLimit   uint64
+	gasTipCap  *big.Int
+	gasFeeCap  *big.Int
+	accessList types.AccessList
+}
+
+var _ TxCommonWithProto = (*DynamicFeeTx)(nil)
+
+func (tx *DynamicFeeTx) Version() uint32 {
+	return DynamicFeeTxType
+}
+
+func (tx *DynamicFeeTx) ChainID() uint32 {
+	return tx.chainID
+}
+
+func (tx *DynamicFeeTx) Nonce() uint64 {
+	return tx.nonce
+}
+
+func (tx *DynamicFeeTx) Gas() uint64 {
+	return tx.gasLimit
+}
+
+func (tx *DynamicFeeTx) GasTipCap() *big.Int {
+	return tx.gasTipCap
```

Comment:
`return new(big.Int).Set(tx.gasTipCap)`
b/c the return value might do Mul() or Add() operation, which will change its value

Reply:
fixed